### ISSUE TITLE
Replace Fortran FRE tools with Python ones

### ISSUE THREAD

**Issue by ashjbarnes** [COSIMA/regional-mom6#68]
A barrier for new users on a new machine is the package's reliance on having a compiled set of GFDL's FRE tools. There are Python tools that exist which could help with mosaic generation [here](https://github.com/ESMG/gridtools/blob/de0a18c1ce0807748aa70023300dfc415277bd4c/gridtools/grids/mom6.py) but I'm not sure about the processor mask. 

Ideally we should add this package as a dependency and move away from FRE tools. This package also includes some great grid generation tools which we should include as options in our package

**Comment by navidcy** [comment]
Being depended on FRE tools also limits are CI-testing capabilities. Unless there is a way to run FRE tools somehow that I don't know. @angus-g?

**Comment by ashjbarnes** [comment]
Yeah for sure it's really annoying! Seems like we're only missing the check mask FRE tool which generates the processor mask. Others appear to be present in the grid tools repo

**Comment by angus-g** [comment]
> Being depended on FRE tools also limits are CI-testing capabilities. Unless there is a way to run FRE tools somehow that I don't know. @angus-g?

It's pretty easy to build them if we want to test things as they are, or make sure any Python implementations are equivalent. Let me know if you want that set up.

**Comment by ashjbarnes** [comment]
True - whatever you think is best Angus! Is it easy enough to include the compiling of Fortran scripts in the regular conda / pip installation process? Since the python implementation is missing the check-mask maybe it makes sense to stick with Fortran for now and include it in the installation pipeline

**Comment by ashjbarnes** [comment]
Moving forward, the default for the COSIMA community will be the NUOPC workflow, which doesn't need FRE tools at all if you use auto masking. For those who want to use manual PE masking, FRE tools are being supported by ACCESS-NRI, but there are no plans to build them in python as far as I know. 

There would be little benefit in replacing FRE tools with python ones given most people won't need them. Those who do can still use the originals FORTRAN ones though!
